htdocs/product/stock/class/mouvementstock.class.php

```diff
@@ -348,7 +357,7 @@ public function _create($user, $fk_product, $entrepot_id, $qty, $type, $price =
 						$i++;
 					}
 				}
-				else   // If not found, we add record
+				elseif ($type == 0 || $type == 3)   // If not found, we add record
```

Comment:
This break the autorepair feature.
When we have in stock a record with batch number X, when we change stock of X (and we must be able since it is into database), we refuse because the record into Productlot is not correctly synchronized.
We have no reason to add a condition here. If a record is in stock with a lot X, we must create to fix consistency.
This will generate a dead lock situation.

Reply:
@altairis-tof

Reply:
@lmag i saw it but did not understand what to do...
@eldy do i have to just remove this code or do it another way ? i'm pulling my hair out ;-)

Reply:
@eldy ok i have removed the test. you can merge (i hope)